temporian/implementation/numpy_cc/operators/calendar/common.h

```diff
@@ -0,0 +1,51 @@
+#include "absl/time/time.h"
+
+static absl::TimeZone handle_tz(const pybind11::object tz) {
+  if (pybind11::isinstance<pybind11::int_>(tz)) {
+    // handle tz as an int
+    int int_tz = tz.cast<int>();
+    // tz is expresed in hours, needs to be converted to seconds
+    return absl::FixedTimeZone(int_tz * 60 * 60);
+
+  } else if (pybind11::isinstance<pybind11::float_>(tz)) {
+    // handle tz as an float
+    float float_tz = tz.cast<float>() * 60 * 60;
+    int int_tz = static_cast<int>(std::round(float_tz));
+    // tz is expresed in hours, needs to be converted to seconds
+    return absl::FixedTimeZone(int_tz);
+
+  } else if (pybind11::isinstance<pybind11::str>(tz)) {
+    // handle tz as an string
+    std::string str_tz = tz.cast<std::string>();
+    absl::TimeZone parsed_tz;
+    if (!absl::LoadTimeZone(str_tz, &parsed_tz)) {
+      throw std::invalid_argument("Invalid timezone '" + str_tz +
```

Comment:
Currently, we don't use exceptions. Would there be another way to pass the error?

Reply:
I like having exceptions in these cases and I think we have some in python, is there any reason to avoid the exceptions in c++?

I see a couple of options:

1. change it to a warning and ignore the timezone (use UTC)
2. return some kind of error code and handle it on python

I think the exception is better than the warning but that's a design decision and if you think the warning fits better with our current design I'm not against it.

number 2 I don't see it as an improvement, it would just move the problem from c++ to python and complicate the return types

Reply:
Our c++ code is not written with exceptions in mind. Before starting to use them, we would have to figure out the implications for all the current and future code. In addition, Google engineers don't use exceptions (https://google.github.io/styleguide/cppguide.html#Exceptions), and since this code is used in Google, it would be a pain to enable them.

Reply:
I changed the implementation to return an error message

Reply:
What about creating a typdef std::optional<std::string> Error ?
This way, it is easier to understand that if empty, there is not error.
As discussed, the alternative is to use absl::Status: https://abseil.io/docs/cpp/guides/status

Reply:
My first option was using the Status but I ran into a couple of problems:
1. Our abseil version is quite old, had to update it
2. Had to add [pybind11_abseil](https://github.com/pybind/pybind11_abseil) in order to convert the `absl::Status `to a python object
3. `pybind11_abseil` needed a newer version of `rules_python` which triggers many errors in other dependencies which should also be updated (protobuf for instance)

Given this I went with your other option and use `std::optional<sdt::string>` will push the changes soon